MongoRepository/Repository/MongoRepository.cs

```diff
@@ -155,6 +191,20 @@ public virtual void Update(IEnumerable<T> entities)
                 this.collection.ReplaceOne(GetIDFilter(entity.Id), entity, new UpdateOptions { IsUpsert = true });
         }
 
+
+
+        /// <summary>
+        /// Upserts the entities asynchronously
+        /// </summary>
+        /// <param name="entities">The entities to update</param>
+        public virtual async Task UpdateAsync(IEnumerable<T> entities)
+        {
+            foreach (T entity in entities)
+                await this.collection.ReplaceOneAsync(GetIDFilter(entity.Id), entity, new UpdateOptions { IsUpsert = true });
```

Comment:
Same thing here instead of calling ReplaceOne can't we directly call UpdateManyAsync which will make it a batch opration instead of calling ReplaceOne on every entity??